EXECUTION_SWEEP cannot change the status of execution correctly

**Expected behavior and actual behavior:**
Expected: When the status of all tasks is a success, the status of execution should also be modified to success
Actual: When the status of all tasks is a success, the status of execution has not been modified to success

**Versions:**
Please specify the versions of following systems.

- harbor version: v2.8.0

error.log
```
2023-03-23T05:00:01Z [ERROR] [/pkg/task/sweep_job.go:94]: failed to fix dangling state executions, error: failed to fix dangling state execution: ERROR: syntax error at or near "$1" (SQLSTATE 42601)
```

--- a/src/pkg/task/sweep_manager.go
+++ b/src/pkg/task/sweep_manager.go
@@ -200,11 +200,10 @@ func (sm *sweepManager) FixDanglingStateExecution(ctx context.Context) error {
             ELSE 'Success'
             END
 WHERE status = 'Running'
-  AND start_time < now() - INTERVAL ?
+  AND EXTRACT(epoch FROM NOW() - start_time)/3600 >  ?
   AND NOT EXISTS (SELECT 1 FROM task WHERE execution_id = execution.id AND status = 'Running')`
 
-	intervalStr := fmt.Sprintf("%d hour", config.MaxDanglingHour())
-	_, err = ormer.Raw(sql, intervalStr).Exec()
+	_, err = ormer.Raw(sql, config.MaxDanglingHour()).Exec()
 	if err != nil {
 		return errors.Wrap(err, "failed to fix dangling state execution")
 	}

--- a/src/pkg/task/sweep_manager_test.go
+++ b/src/pkg/task/sweep_manager_test.go
@@ -19,14 +19,16 @@ import (
 	"testing"
 	"time"
 
+	"github.com/stretchr/testify/suite"
+
 	"github.com/goharbor/harbor/src/lib/errors"
 	"github.com/goharbor/harbor/src/pkg/task/dao"
+	htesting "github.com/goharbor/harbor/src/testing"
 	"github.com/goharbor/harbor/src/testing/mock"
-	"github.com/stretchr/testify/suite"
 )
 
 type sweepManagerTestSuite struct {
-	suite.Suite
+	htesting.Suite
 	execDao *mockExecutionDAO
 	mgr     *sweepManager
 }
@@ -36,6 +38,7 @@ func TestSweepManager(t *testing.T) {
 }
 
 func (suite *sweepManagerTestSuite) SetupSuite() {
+	suite.Suite.SetupSuite()
 	suite.execDao = &mockExecutionDAO{}
 	suite.mgr = &sweepManager{execDAO: suite.execDao}
 }
@@ -53,4 +56,9 @@ func (suite *sweepManagerTestSuite) TestGetCandidateMaxStartTime() {
 	startTime, err = suite.mgr.getCandidateMaxStartTime(context.TODO(), "WEBHOOK", 1, 10)
 	suite.NoError(err, "should not got error")
 	suite.Equal(now.String(), startTime.String())
+}
+
+func (suite *sweepManagerTestSuite) Test_sweepManager_FixDanglingStateExecution() {
+	err := suite.mgr.FixDanglingStateExecution(suite.Context())
+	suite.NoError(err, "should not got error")
 }

